apps/aeprimop/src/aeprimop.erl

```diff
@@ -566,13 +578,31 @@ lock_namefee(Kind, From, Amount, #state{protocol = Protocol} = S) ->
 
 spend_fee_op(From, Amount) when ?IS_HASH(From),
                                 ?IS_NON_NEG_INTEGER(Amount) ->
-    {spend_fee, {From, Amount}}.
-
-spend_fee({From, Amount}, #state{} = S) when is_integer(Amount), Amount >= 0 ->
-    {Sender1, S1}   = get_account(From, S),
-    assert_account_balance(Sender1, Amount),
-    {ok, Sender2}   = aec_accounts:spend_without_nonce_bump(Sender1, Amount),
-    put_account(Sender2, S1).
+    {spend_fee, {From, Amount, 0}}.
+
+spend_fee_op(From, DelegatedAmount, Amount)
+        when ?IS_HASH(From), ?IS_NON_NEG_INTEGER(DelegatedAmount + Amount) ->
```

Comment:
Actually, one would like to enforce the stronger ?IS_NON_NEG_INTEGER(DelegateAmount), ?IS_NON_NEG_INTEGER(Amount).

Reply:
Yes 👍